tools/clang/unittests/HLSL/PixTest.cpp

```diff
@@ -1640,6 +1645,130 @@ TEST_F(PixTest, PixDebugCompileInfo) {
 }
 
+CComPtr<IDxcBlob> PixTest::Compile(const char* hlsl, const wchar_t* profile)
+{
+
+  CComPtr<IDxcLibrary> pLib;
+  VERIFY_SUCCEEDED(m_dllSupport.CreateInstance(CLSID_DxcLibrary, &pLib));
+
+  CComPtr<IDxcCompiler> pCompiler;
+  CComPtr<IDxcCompiler2> pCompiler2;
+
+  CComPtr<IDxcOperationResult> pResult;
+  CComPtr<IDxcBlobEncoding> pSource;
+  CComPtr<IDxcBlob> pProgram;
+  CComPtr<IDxcBlob> pPdbBlob;
+  WCHAR *pDebugName = nullptr;
+
+  VERIFY_SUCCEEDED(CreateCompiler(&pCompiler));
+  VERIFY_SUCCEEDED(pCompiler.QueryInterface(&pCompiler2));
+  CreateBlobFromText(hlsl, &pSource);
+  LPCWSTR args[] = {L"/Zi", L"/Qembed_debug"};
+  VERIFY_SUCCEEDED(pCompiler2->CompileWithDebug(
+      pSource, L"source.hlsl", L"main", profile, args, _countof(args),
+      nullptr, 0, nullptr, &pResult, &pDebugName, &pPdbBlob));
+
+  HRESULT hr;
+  VERIFY_SUCCEEDED(pResult->GetStatus(&hr));
+  if (FAILED(hr))
+  {
+    CComPtr<IDxcBlobEncoding> pErrors;
+    pResult->GetErrorBuffer(&pErrors);
+
+    if (pErrors && pErrors->GetBufferSize() > 0) {
+      auto errorMessages =
+          reinterpret_cast<const char *>(pErrors->GetBufferPointer());
+      std::wstring wideErrrors;
+      wideErrrors.assign(errorMessages,
+                         errorMessages + pErrors->GetBufferSize());
+      WEX::Logging::Log::Comment(L"Compilation failed. Error messages:");
+      WEX::Logging::Log::Comment(wideErrrors.c_str());
+    }
+
+    VERIFY_SUCCEEDED(hr);
+  }
+  VERIFY_SUCCEEDED(pResult->GetResult(&pProgram));
+
+  return pProgram;
+}
+
+std::string PixTest::RunShaderAccessTrackingPassAndReturnOutputMessages(IDxcBlob* blob)
+{
+  CComPtr<IDxcBlob> dxil = FindModule(DFCC_ShaderDebugInfoDXIL, blob);
+  CComPtr<IDxcOptimizer> pOptimizer;
+  VERIFY_SUCCEEDED(
+      m_dllSupport.CreateInstance(CLSID_DxcOptimizer, &pOptimizer));
+  std::vector<LPCWSTR> Options;
+  Options.push_back(L"-opt-mod-passes");
+  Options.push_back(L"-hlsl-dxil-pix-shader-access-instrumentation,config=,checkForDynamicIndexing=1");
+
+  CComPtr<IDxcBlob> pOptimizedModule;
+  CComPtr<IDxcBlobEncoding> pText;
+  VERIFY_SUCCEEDED(pOptimizer->RunOptimizer(
+      dxil, Options.data(), Options.size(), &pOptimizedModule, &pText));
+
+  std::string outputText;
+  if (pText->GetBufferSize() != 0) {
+    outputText = reinterpret_cast<const char *>(pText->GetBufferPointer());
+  }
+
+  return outputText;
+}
+
+TEST_F(PixTest, CheckSATPassFor66_NoDynamicAccess) {
+
+  const char *noDynamicAccess = R"(
+    [RootSignature("")]
+    float main(float pos : A) : SV_Target {
+      float x = abs(pos);
+      float y = sin(pos);
+      float z = x + y;
+      return z;
+    }
+  )";
+
+  auto compiled = Compile(noDynamicAccess, L"ps_6_6");
+  auto satResults = RunShaderAccessTrackingPassAndReturnOutputMessages(compiled);
+  VERIFY_IS_TRUE(satResults.empty());
+}
+
+TEST_F(PixTest, CheckSATPassFor66_DynamicFromRootSig) {
+
+  const char *noDynamicAccess = R"x(
```

Comment:
May want to change the variable name as this does have dynamic access

Reply:
Thanks. Fixed.